src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -166,6 +221,32 @@ public void DatagramSent()
             Interlocked.Increment(ref _datagramsSent);
         }
 
+        private static string GetErrorType(SocketError socketError) => socketError switch
+        {
+            // Common connect() errors expected to be seen:
+            // https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-connect#return-value
+            // https://man7.org/linux/man-pages/man2/connect.2.html
+            SocketError.NetworkDown => "network_down",
+            SocketError.AddressAlreadyInUse => "address_already_in_use",
+            SocketError.Interrupted => "interrupted",
+            SocketError.InProgress => "in_progress",
+            SocketError.AlreadyInProgress => "already_in_progress",
+            SocketError.AddressNotAvailable => "address_not_available",
+            SocketError.AddressFamilyNotSupported => "address_family_not_supported",
+            SocketError.ConnectionRefused => "connection_refused",
+            SocketError.Fault => "fault",
+            SocketError.InvalidArgument => "invalid_argument",
+            SocketError.IsConnected => "is_connected",
+            SocketError.NetworkUnreachable => "network_unreachable",
+            SocketError.HostUnreachable => "host_unreachable",
+            SocketError.NoBufferSpaceAvailable => "no_buffer_space_available",
+            SocketError.TimedOut => "timed_out",
+            SocketError.AccessDenied => "access_denied",
+            SocketError.ProtocolType => "protocol_type",
+
+            _ => "_OTHER"
```

Comment:
Should this return `socketError.ToString()` instead, similar to how we fall back to exception types?

Reply:
I think reporting an enum member name without converting it to snake case would go against conventions. I'm trying to catch all SocketErrors that can reasonably occur here, users should not see `_OTHER`.